cmd/skopeo/inspect.go

```diff
@@ -85,9 +96,25 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 		return err
 	}
 
-	src, err := parseImageSource(ctx, opts.image, imageName)
-	if err != nil {
-		return fmt.Errorf("Error parsing image name %q: %v", imageName, err)
+	if _, err := alltransports.ParseImageName(imageName); err != nil {
+		return err
+	}
+
+	if err := retryIfNecessary(ctx, func() error {
+		source, err := parseImageSource(ctx, opts.image, imageName)
+		src = source
+		if err != nil {
+			return errors.Wrapf(err, "Error parsing image name %q", imageName)
+		}
+
+		manifest, _, err := src.GetManifest(ctx, nil)
+		rawManifest = manifest
+		if err != nil {
+			return errors.Wrapf(err, "Error retrieving manifest for image")
+		}
+		return err
+	}, opts.retryOpts, opts.global.commandTimeout); err != nil {
+		return err
 	}
 
 	defer func() {
```

Comment:
This should happen just after a successful `parseImageSource`, so I guess we need two `retryIfNecessary` blocks instead of the one above.

Reply:
one retry block wrapping `parseImageSource` and defer Close(), and the other retry block for `GetManifest`?

Reply:
`defer` only defers until the end of the function, which in this case would be just the retry block, and `src` would not be usable afterwards.

Retry { `src = parseImageSource` }`; `defer src.Close`; Retry { `GetManifest` }.